### ISSUE TITLE
Running without huggingface token cache raises an error

### ISSUE THREAD

**Issue by sigridjineth** [EricLBuehler/candle-vllm#76]
## Description
When trying to run candle-vllm with a Hugging Face model, the program fails if the `~/.cache/huggingface/token` file doesn't exist. It appears that the program is not correctly handling the case where the token file or its parent directories don't exist.

## Steps to Reproduce
1. Ensure `~/.cache/huggingface/token` doesn't exist
2. Run candle-vllm with a command like:
```
cargo run --release -- --port 2000 --model-id meta-llama/Llama-2-7b-chat-hf llama
```
3. Observe the error

## Current Behavior
The program fails with an error: `Error: APIError { data: "No such file or directory (os error 21)" }`

## Expected Behavior
The program should:
1. Create the `~/.cache/huggingface` directory if it doesn't exist
2. Create an empty `token` file if it doesn't exist
3. Prompt the user to enter their Hugging Face token if the file is empty
4. Alternatively, provide a clear error message instructing the user to create the token file

## Workaround
Currently, users need to manually create the directory and file:

```
mkdir -p ~/.cache/huggingface
touch ~/.cache/huggingface/token
vim ~/.cache/huggingface/token  # To add the token manually
```

## Additional Context
This issue affects the user experience, especially for new users who may not be familiar with the expected file structure. Implementing proper file and directory handling would improve the ease of use of candle-vllm.

Thanks for awesome project for rescuing vllm from python!

**Comment by guoqingbao** [comment]
> ## Description
> When trying to run candle-vllm with a Hugging Face model, the program fails if the `~/.cache/huggingface/token` file doesn't exist. It appears that the program is not correctly handling the case where the token file or its parent directories don't exist.
> 
> ## Steps to Reproduce
> 1. Ensure `~/.cache/huggingface/token` doesn't exist
> 2. Run candle-vllm with a command like:
> 
> ```
> cargo run --release -- --port 2000 --model-id meta-llama/Llama-2-7b-chat-hf llama
> ```
> 
> 3. Observe the error
> 
> ## Current Behavior
> The program fails with an error: `Error: APIError { data: "No such file or directory (os error 21)" }`
> 
> ## Expected Behavior
> The program should:
> 
> 1. Create the `~/.cache/huggingface` directory if it doesn't exist
> 2. Create an empty `token` file if it doesn't exist
> 3. Prompt the user to enter their Hugging Face token if the file is empty
> 4. Alternatively, provide a clear error message instructing the user to create the token file
> 
> ## Workaround
> Currently, users need to manually create the directory and file:
> 
> ```
> mkdir -p ~/.cache/huggingface
> touch ~/.cache/huggingface/token
> vim ~/.cache/huggingface/token  # To add the token manually
> ```
> 
> ## Additional Context
> This issue affects the user experience, especially for new users who may not be familiar with the expected file structure. Implementing proper file and directory handling would improve the ease of use of candle-vllm.
> 
> Thanks for awesome project for rescuing vllm from python!

Thanks for raising this issue, we will address it ASAP.

**Comment by guoqingbao** [comment]
This issue has been addressed in #81
